sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/RestrictDeserializedTypes.cs

```diff
@@ -103,11 +103,50 @@ public override ProgramState ObjectCreated(ProgramState programState, SymbolicVa
                     {
                         programState = programState.SetConstraint(symbolicValue, SerializationConstraint.Unsafe);
                     }
+
+                    if (IsLosFormatter(typeSymbol) &&
+                        !HasSafeConstructorInitialization(objectCreation, programState))
+                    {
+                        // For LosFormatter the rule is raised directly on the constructor.
+                        addNode(objectCreation);
+                    }
                 }
 
                 return base.ObjectCreated(programState, symbolicValue, instruction);
             }
 
+            private bool HasSafeConstructorInitialization(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                // The constructor is safe only if it has 2 arguments and the first argument value is true.
+                if (objectCreation.ArgumentList.Arguments.Count != 2)
+                {
+                    return false;
+                }
+
+                var firstArgument = objectCreation.ArgumentList.Arguments[0].Expression;
```

Comment:
Contrary to `JavaScriptSerializer` constructor that was taking a single argument, or to formatters that had initializer, here the constructor can take 2 arguments, so it would be better to first check for named argument, in case the expressions were swapped for some reason. That way we will also cover the case:
```
new LosFormatter (macKeyModifier: new byte[0], enableMac: false)
```

Reply:
That's a very nice case I didn't think of. Thanks!